File: kwetter-dao/src/main/kotlin/com/zwolsman/kwetter/dao/models/KweetEntities.kt
package com.zwolsman.kwetter.dao.models

data class KweetEntities (val hashtags: List<Hashtag> = emptyList(), val userMentions:List<UserMention> = emptyList(), val urls: List<Url> = emptyList())

data class Hashtag(val range: Array<Int>, val text: String)

data class UserMention(val range: Array<Int>, val name: String)

data class Url(val range: List<Int>, val url: String, val expandedUrl: String) {
    val displayUrl: String
        get() = expandedUrl.substringAfter("")
}

File: kwetter-ios/kwetter-ios/models/KweetEntities.swift
import Foundation
struct KweetEntities : Codable {
	let hashtags : [Hashtag]
	let userMentions : [UserMention]
	let urls : [Url]
}

struct Hashtag : Codable {
    let range : [Int]
    let text : String
}

struct UserMention : Codable {
    let range: [Int]
    let name: String
}

struct Url : Codable {
    let url : String
    let expandedUrl: String
    let displayUrl: String
}
